coral-spark/src/main/java/com/linkedin/coral/spark/transformers/TransportableUDFTransformer.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2018-2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.spark.transformers;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.Set;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.spark.sql.SparkSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linkedin.coral.common.transformers.SqlCallTransformer;
+import com.linkedin.coral.hive.hive2rel.functions.VersionedSqlUserDefinedFunction;
+import com.linkedin.coral.spark.containers.SparkUDFInfo;
+import com.linkedin.coral.spark.exceptions.UnsupportedUDFException;
+
+
+/**
+ * This transformer transforms legacy Hive UDFs to an equivalent Transportable UDF.
+ * Check `CoralSparkTest#testTransportableUDFTransformer()` for example.
+ */
+public class TransportableUDFTransformer extends SqlCallTransformer {
+  private final String hiveUDFClassName;
+  private final String sparkUDFClassName;
+  private final String artifactoryUrlSpark211;
+  private final String artifactoryUrlSpark212;
+  private final Set<SparkUDFInfo> sparkUDFInfos;
+  private ScalaVersion scalaVersion;
+
+  public TransportableUDFTransformer(String hiveUDFClassName, String sparkUDFClassName, String artifactoryUrlSpark211,
+      String artifactoryUrlSpark212, Set<SparkUDFInfo> sparkUDFInfos) {
+    this.hiveUDFClassName = hiveUDFClassName;
+    this.sparkUDFClassName = sparkUDFClassName;
+    this.artifactoryUrlSpark211 = artifactoryUrlSpark211;
+    this.artifactoryUrlSpark212 = artifactoryUrlSpark212;
+    this.sparkUDFInfos = sparkUDFInfos;
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(TransportableUDFTransformer.class);
+  public static final String DALI_UDFS_IVY_URL_SPARK_2_11 =
+      "ivy://com.linkedin.standard-udfs-dali-udfs:standard-udfs-dali-udfs:2.0.3?classifier=spark_2.11";
+  public static final String DALI_UDFS_IVY_URL_SPARK_2_12 =
+      "ivy://com.linkedin.standard-udfs-dali-udfs:standard-udfs-dali-udfs:2.0.3?classifier=spark_2.12";
+
+  public enum ScalaVersion {
+    SCALA_2_11,
+    SCALA_2_12
+  }
+
+  @Override
+  protected boolean condition(SqlCall sqlCall) {
```

Comment:
does it make sense to add another condition - 
`sqlCall.getOperator() instanceof  VersionedSqlUserDefinedFunction` ?
We seem to be assuming that in the transform()